board/flags-config.cmake

```diff
@@ -0,0 +1,74 @@
+# This function provides a set of specific flags for each supported board
+# Depending on the processor type. The objective is to optimize for size.
+# Thus, all of the fllowing flags are chosen carefully to reduce binary 
+# footprints.
+
+# Set generic minimization flags for all platforms.
+# These flags are the same for all cross-compilation cases.
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Os -fdata-sections -ffunction-sections")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fomit-frame-pointer -fno-unwind-tables")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-asynchronous-unwind-tables -fvisibility=hidden")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fshort-enums -finline-small-functions")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -findirect-inlining -fno-common") 
+#-flto -fuse-ld=gold # There is an issue with gold link when compiling on 
+# Ubuntu 16. At that point gcc linker did not integrate the flto support 
+# inside and it was a separate plugin that need to be added. Therefore, 
+# this can be added when mlpack Azure CI moves toward Ubuntu 20.
+
+option(RPI0 "Optimize compiler flags for Raspberry PI 0." OFF)
+option(RPI1 "Optimize compiler flags for Raspberry PI 1." OFF)
+option(RPI2 "Optimize compiler flags for Raspberry PI 2." OFF)
+option(RPI3 "Optimize compiler flags for Raspberry PI 3." OFF)
+option(RPI4 "Optimize compiler flags for Raspberry PI 4." OFF)
+option(BV "Optimize compiler flags for Beagleboard V." OFF)
+option(JETSONAGX "Optimize compiler flags for Nvidia Jetson AGX Xavier." OFF)
+option(KATAMI "Optimize compiler flags for Pentium 3 Katami processors." OFF)
+option(COPPERMINE "Optimize compiler flags for Pentium 3 Coppermine processors." OFF)
+option(NORTHWOOD "Optimize compiler flags for Pentium 4 Northwood processors." OFF)
```

Comment:
Personally I think this is better as a single string option like `BOARD_NAME`, like you had it before.  The reason I say that is because with ten different `option()`s, the user could set all of them to true, but they have no way of knowing from the documentation what the precedence order is.  Since it doesn't make sense to set multiple boards (and since you are using an `if`/`elseif` structure below), I think it would be a lot cleaner to just use `BOARD_NAME` like before. :+1:

Reply:
Ok, I will change that back, I just hope that the user does not misspell the board name

Reply:
Well, you can catch that situation pretty easily with the `if`/`elseif` block; just add an `else` at the end if the board isn't recognized. :)